pp.c

```diff
@@ -7253,6 +7253,75 @@ PP(pp_unweaken)
     RETURN;
 }
 
+PP(pp_blessed)
+{
+    dSP;
+    SV *arg = TOPs;
+    SV *rv;
+
+    SvGETMAGIC(arg);
+
+    if(!SvROK(arg) || !SvOBJECT((rv = SvRV(arg)))) {
+        SETs(&PL_sv_undef);
+        RETURN;
+    }
+
+    if((PL_op->op_private & OPpTRUEBOOL) ||
+            ((PL_op->op_private & OPpMAYBE_TRUEBOOL) && (block_gimme() == G_VOID))) {
+        /* refs are always true, but we just have to check for the annoying
+         * cornercase of the package named "0" */
```

Comment:
The truthiness of refs isn't really relevant (neither here nor in `pp_ref`). How about something like `all package names are true, except the annoying corner case of "0"`?